### DISCUSSION TITLE
Any solution for retry an argo workflow from a specified steps.

### DISCUSSION THREAD

**Discussion by jiangfwa** [argoproj/argo-workflows#7534] [Q&A] [answered]
Is there a way to retry a failed workflow from a specified step?

The case is that a workflow failed due to the wrong artifact generated by the early step before the failed step. I clicked the retry button, it always retried the failed step using the same wrong artifact which is hopeless. Is there a possible way to retry the workflow from the _generate_artifact_ step?

For example, set a flag for the _generate_artifact_ step, so that I can retry it. Or put several steps in a step set, then the retry action is acted to the step set.

**Comment by terrytangyuan** [comment, marked as answer]
You can use `argo retry --restart-successful` and `--node-field-selector` to select nodes that need to be re-run when retrying (even when they are successful).

**Reply by jiangfwa** [reply]
The UI is not support this feature, I guess, right? Another words, Is it possible to bind two steps. 

For example, step A is before the step B. Step A succeed, Step B failed. We can setup some kind of retryStrategy to retry both of them? This kind of binding definition can be written in the workflow YAML file.

**Reply by terrytangyuan** [reply]
Yes, you should be able to retry both of them with the trick I mentioned.

**Reply by RoryDoherty** [reply]
@terrytangyuan  say I have a workflow X and in this workflow are the steps A, B, C, D, E. 

A runs first and sets up some infrastructure for steps B C D E. 
```
       A
   ____|____
   |  |  |  |
   B  C  D  E
```
If step C then fails but the rest succeed is there any way to just run A then C again?

I tried running
```
argo retry X --restart-successful --node-field-selector name=A
```
but it ran all of the child steps again

**Reply by terrytangyuan** [reply]
Remove restart-successful flag.

**Reply by RoryDoherty** [reply]
Thanks for the quick reply,
I just tried that but it did not run step A again, it only ran the failed step C again. 
Step C relies on step A setting up some infrastructure for it to use 

I ran 
```
argo retry X --node-field-selector name=A
```
i'm on Argo Workflows 3.3.5 by the way

**Reply by terrytangyuan** [reply]
This PR might fix your issue. Stay tuned: https://github.com/argoproj/argo-workflows/pull/9028

**Reply by RoryDoherty** [reply]
@terrytangyuan I've upgraded to Argo Workflows 3.4.2 and just tried the above scenario again.
With a DAG and steps that have depends if I click Retry in the UI then only the failed step is retried.
The step that it depends on to set up some infrastructure did not get run again

**Reply by tooptoop4** [reply]
did u raise new issue @RoryDoherty ?

**Reply by RoryDoherty** [reply]
@tooptoop4 I have not raised a new issue yet but I will today as this is impacting us daily

**Reply by RoryDoherty** [reply]
Issue created https://github.com/argoproj/argo-workflows/issues/10425

**Comment by wesleyscholl** [comment]
We are also experiencing this issue. When retrying after failure, the workflow hangs in "Running" mode.
